File: Toll-Calculator/Models/Vehicles/Tractor.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Toll_Calculator.Enums;
using Toll_Calculator.Interfaces;

namespace Toll_Calculator.Models.Vehicles
{
    public class Tractor : IVehicle
    {
        public bool IsTollFree()
        {
            return true;
        }

        public VehicleType GetVehicleType()
        {
            return VehicleType.Tractor;
        }
    }
}

File: Golang/vehicle/tractor.go
package vehicle

// tractor struct for handling Tractor type.
type tractor struct{}

// GetVehicleType will return the "Tractor" type.
func (t tractor) GetVehicleType() string {
	return "Tractor"
}

// IsTollFree return true for Tractor type.
func (t tractor) IsTollFree() bool {
	return true
}

// NewTractor return a vehicle type of Tractor.
func NewTractor() Vehicle {
	return &tractor{}
}
